src/tree/mod.rs

```diff
@@ -1134,6 +1248,10 @@ impl<T> Tree<T> {
     }
 }
 
+trait IteratorNew<'a, T, I> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> I;
+}
+
```

Comment:
I'll see if I can re-visit this so that I don't have to do the trait workaround.  I just don't want them public from the library perspective.  Internally, I don't really care where they are (although I was trying to keep them in their own file because the tree module is getting large).